### PR TITLE
CableReady 5.0 installer

### LINKED ISSUE #130: cable_ready:channel generator has some issues
1. `rails destroy cable_ready:channel Sailor` does not seem to destroy anything.
2. It can't seem to locate the Stimulus controller template?

In this case I'd previously run the generator with "y" to `broadcast_to`, "Sailor" for the resource name, and "y" for the Stimulus controller. It complained about not finding the controller template in any of my source paths, as shown below.

```
~/testcr  $ rails destroy cable_ready:channel Sailor
Running via Spring preloader in process 69822
    generate  channel
       rails  generate channel sailor
Are you streaming to a resource using broadcast_to? (y/N) y
Which resource are you streaming for? (Sailor) 
Are you going to use a Stimulus controller to subscribe to this channel? (y/N) y
Could not find "app/javascript/controllers/%file_name%_controller.js" in any of your source paths. Your current source paths are: 
/Users/leastbad/.rvm/gems/ruby-2.6.5/gems/cable_ready-5.0.0.pre0/lib/generators/cable_ready/templates
```

Happy to help troubleshoot. Let me know how I can be of assistance.

### LINKED ISSUE #131: Specify CableReadyBroadcastJob queue in initializer
1. It seems like the `queue_as` value in the job class should default to `:default` but be customizable via the CR initializer.
2. Following this logic a bit further, what if we added a keyword parameter `queue` to the `broadcast_later` and `broadcast_later_to` method signature that could override?

`broadcast_later(queue: "cable_ready")`

What do you think, @julianrubisch?

### LINKED ISSUE #180: Helper generator Vite problems
The helpers_generator here fails with Vite:
https://github.com/stimulusreflex/cable_ready/blob/1bded053b64e8f497fcf368a258b0326ea4c886f/lib/generators/cable_ready/helpers_generator.rb#L12-L22
It assumes the `main_folder` is "app/javascript" and the entrypoint directory is named "packs". 

A slight adjustment makes it work for either:
```diff
    main_folder = defined?(Webpacker) ? Webpacker.config.source_path.to_s.gsub("#{Rails.root}/", "") : "app/javascript"
++  main_folder = "app/frontend" if defined? ViteRuby

    filepath = [
      "#{main_folder}/controllers/index.js",
      "#{main_folder}/controllers/index.ts",
      "#{main_folder}/packs/application.js",
      "#{main_folder}/packs/application.ts",
++    "#{main_folder}/entrypoints/application.js",
++    "#{main_folder}/entrypoints/application.ts"
    ]
```
There are some similar install tasks in SR that also break in Vite but just need a very slightly different path. And there's this Webpacker-specific line that gets added during setup:
```js
StimulusReflex.debug = process.env.RAILS_ENV === 'development'
```
which throws an error with Vite. It works with a slight (Vite-specific) adjustment:
```diff
--StimulusReflex.debug = process.env.RAILS_ENV === 'development'
++StimulusReflex.debug = import.meta.env.DEV
```

I'm wondering if these could be generalised a bit. I haven't tried with `jsbundling-rails`, but it also uses ESBuild so I assume it would be similar to Vite in regards to default paths..?

_Originally posted by @Matt-Yorkley in https://github.com/stimulusreflex/cable_ready/issues/179#issuecomment-1028036454_

### PR DESCRIPTION
**Pull request by leastbad**
# Type of PR

Feature

## Description

This PR is the CableReady counterpart to https://github.com/stimulusreflex/stimulus_reflex/pull/610.

I'm not going to rehash everything discussed in the SR installer; it is a modified subset of the same system. There are some small differences but great pains have been taken to keep everything consistent. **It is possible to install CableReady, and then add StimulusReflex to the project and install StimulusReflex at a later date.** Note that if you try to run the CableReady installer when StimulusReflex is in your project, it will exit and tell you to run the SR installer; 

It supports all major bundlers in their default configuration:

Rails 6.1: webpacker 5.4.3, shakapacker, vite
Rails 7: importmap, esbuild, shakapacker, vite

I've adapted the `cable_ready:channel` generator to work with the new installer. The `cable_ready:initializer` generator has been removed in favor of the installer.

```bash
rails cable_ready:install # fully automatic detections, with prompts
rails cable_ready:install esbuild # specify a bundler
rails cable_ready:install entrypoint=app/frontend webpacker # install with webpacker 5.4.3 in app/frontend
rails cable_ready:install:step action_cable yarn bundle # re-run one or more steps
rails cable_ready:install:restart # wipe state and restart install

rails cable_ready:install uncomment=true # uncomment any necessary includes that have been commented
rails cable_ready:install spring=false # do not remove spring (equivalent to answering `n`)
rails cable_ready:install trace=true # debug install process with full stack trace

rails cable_ready:install:step mrujs # install mrujs
rails cable_ready:install:step compression # monkey patch Action Cable to compress WS traffic with deflate


# full fat install with zero interactive prompts, suitable for scripting
rails cable_ready:install esbuild entrypoint=app/javascript uncomment=true spring=true
```
## Open questions and stuff still to be done

1. How does the installer play with JumpStart and BulletTrain?

## Why should this be added

We really need to up our installation and integration game to overcome the emotional barrier people often feel when comparing us against a built-in solution. The more we can make installing CR feel like ordering a Big Mac combo, the less water "golden path" arguments hold.

Closes #179 with massive kudos to @julianrubisch for showing me how to structure an installer.
Closes #180 with apologies to @Matt-Yorkley for the delay
Fixes #130 with a hat-tip to @julianrubisch 
Fixes #131

## Checklist

- [x] My code follows the style guidelines of this project
- [x] Checks (StandardRB & Prettier-Standard) are passing

### CODE REVIEW

File: lib/generators/cable_ready/templates/app/controllers/examples_controller.rb.tt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class ExamplesController < ApplicationController
+  layout false
+
+  def show
+    respond_to do |format|
+      format.html
+      # format.turbo_stream do
+      #   render turbo_stream: turbo_stream.append_all("body", cable_ready_tag(cable_car.console_log(message: "Dear John,\n\nI'd hoped this day would never come. I loved you, but I don't recognize who you've become.\n\nIt's over, and I'm taking the kids.\n\nDon't try to contact me. I hope that Turbo Streams makes you happy in all of the ways that I never could.\n\nGladys")))
```
Comment (julianrubisch) [inline lib/generators/cable_ready/templates/app/controllers/examples_controller.rb.tt:8]: Can't we remove this passive aggressive example completely? @marcoroth

File: lib/install/action_cable.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require "cable_ready/installer"
+
+# verify that Action Cable is installed
+if defined?(ActionCable::Engine)
+  say "✅ ActionCable::Engine is required and in scope"
+else
+  halt "ActionCable::Engine is not required. Please add or uncomment `require \"action_cable/engine\"` to your `config/application.rb`"
```
Comment (julianrubisch) [inline lib/install/action_cable.rb:9]: This ("is not required") could be misunderstood. Suggestion:

```suggestion
  halt "ActionCable::Engine is not loaded. Please add or uncomment `require \"action_cable/engine\"` to your `config/application.rb`"
```

File: lib/install/action_cable.rb [hunk 3]
```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require "cable_ready/installer"
+
+# verify that Action Cable is installed
+if defined?(ActionCable::Engine)
+  say "✅ ActionCable::Engine is required and in scope"
```
Comment (julianrubisch) [inline lib/install/action_cable.rb:7]: See below

```suggestion
  say "✅ ActionCable::Engine is loaded and in scope"
```

File: lib/generators/cable_ready/templates/app/controllers/examples_controller.rb.tt (hunk 1 above)
Comment (marcoroth) [reply lib/generators/cable_ready/templates/app/controllers/examples_controller.rb.tt:8]: Fine for me 👍🏼

**Pull request merged**